### DISCUSSION TITLE
`resources/usr/bin` modules binaries loading not working on Google Cloud

### DISCUSSION THREAD

**Discussion by VasLem** [nextflow-io/nextflow#4939] [Q&A] [answered]
## Bug report 

For the following setup:
```
├── main.nf
├── modules
│   └── download
│       ├── cosmic_annotation.nf
│       ├── resources
│       │   └── usr
│       │       └── bin
│       │           └── read_cosmic.py
│       └── utils.nf
└── nextflow.config
```
with the main.nf being:
```
import {download_cosmic} from './modules/download/cosmic_annotation'
workflow {
download_cosmic()
}
```
and the cosmic_annotation.nf being:
```
process download_cosmic {
script:
"""
read_cosmic.py
"""
}
```

the `read_cosmic.py` file is not found when running on a Google Cloud instance.
The `nextflow.config` contains the line `nextflow.enable.moduleBinaries = true`

(Please follow this template replacing the text between parentheses with the requested information)

### Expected behavior and actual behavior

The python file to be read and executed, but the file is not found. It is neither in the workdir on the GCS.

### Steps to reproduce the problem

`./nextflow run main.nf`

### Program output 

(Copy and paste here output produced by the failing execution. Please highlight it as a code block. Whenever possible upload the `.nextflow.log` file.)

/mnt/disks/mybucket/nextflow/64/37bb5b26c9fd3f24f12bd7c0ebfa7e/.command.sh: line 2: ./read_cosmic.py: No such file or directory

### Environment 

* Nextflow version: 24.03.0-edge

**Comment by pditommaso** [comment]
Is [wave](https://www.nextflow.io/docs/latest/wave.html) enabled in your config?

**Comment by VasLem** [comment]
Hi, no. Is this the only supported way to have such a functionality on Google Batch?

**Comment by pditommaso** [comment, marked as answer]
As per the [documentation](https://www.nextflow.io/docs/latest/module.html#module-binaries) 

> This feature requires the use of a local or shared file system for the pipeline work directory, or [Wave containers](https://www.nextflow.io/docs/latest/wave.html#wave-page) when using cloud-based executors.

**Reply by VasLem** [reply]
Okay thanks, I will  embed the code directly into the process then.
